Spring-server/src/main/java/ikpi63holding/postclient/controllers/ApiController.java

```diff
@@ -29,108 +28,88 @@
 
     private final UserRepository userRepository;
 
-    private final MailboxRepository mailboxRepository;
-
     @Autowired
     private MailServiceFactory mailServiceFactory;
 
     @Autowired
-    ApiController(UserRepository userRepository, MailboxRepository mailboxRepository) {
+    ApiController(UserRepository userRepository) {
         this.userRepository = userRepository;
-        this.mailboxRepository = mailboxRepository;
     }
 
-    @PostMapping("/users/{userId}/mailboxes/{mailboxId}")
-    public void sendMail(@PathVariable Long userId, @PathVariable Long mailboxId,
+    @PostMapping("/users/{userName}/mailboxes/{mailboxId}")
+    public void sendMail(@PathVariable String userName, @PathVariable int mailboxId,
             @RequestBody PostMessage message) throws Exception {
-        User user = userRepository.findById(userId).orElseThrow(
+        User user = userRepository.findByUsername(userName).orElseThrow(
                 () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "No such user"));
 
-        Mailbox mailbox = mailboxRepository.findById(mailboxId).orElseThrow(
-                () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "No such mailbox"));
-
+        Mailbox mailbox = user.getMailboxes().get(mailboxId - 1);
```

Comment:
Isn't that some kind of use of magic numbers?

Reply:
Could be fixed with decrement ("**--mailboxId**")

Reply:
why do we even have to decrement? can we pass a parameter without any additional math?

Reply:
Since it is gefaut list method, we can't really, but this interaction will be reworked, I might even add it in this branch

Reply:
Sure thing